governance/program/src/processor/process_create_signatory_record_from_governance.rs

```diff
@@ -0,0 +1,71 @@
+//! Program state processor
+
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+    rent::Rent,
+    sysvar::Sysvar,
+};
+use spl_governance_tools::account::create_and_serialize_account_signed;
+
+use crate::{
+    error::GovernanceError,
+    state::{
+        enums::GovernanceAccountType,
+        governance::get_governance_required_signatory_data_for_governance,
+        proposal::get_proposal_data_for_governance,
+        signatory_record::{get_signatory_record_address_seeds, SignatoryRecordV2},
+    },
+};
+
+/// Processes CreateSignatoryRecordFromGovernance instruction
+pub fn process_create_signatory_record_from_governance(
```

Comment:
We don't need this instruction and can reuse existing `AddSignatory` instruction with the following changes:

1) Add `Governance` account to the input accounts - this will be a breaking change but the signatory logic is not exposed in the UI and not used so it should be ok
2) If `governance.required_singatories > 0` then fetch optional `required_signatory` account and ensure it matches the `governance` and provided `signatory` 

- `SignatoryRecord` PDA is keyed by the signatory and it would prevent adding the same required signatories - please add test for this 
- The logic in `SignOff` to prevent  signing off when there are not enough required signatories would ensure all required signatories are added - we need a test for this too

Reply:
okay, last commit should have these changes